### DISCUSSION TITLE
source languageserver.golang timeout after 5000ms

### DISCUSSION THREAD

**Discussion by leolin49** [neoclide/coc.nvim#3360] [Q&A] [answered]
Hello, I have configured the go language server gopls in coc-config.json. It works normally in the test program test.go I wrote, it can autocomplete code and jump to definition.
But in large projects, it fails to work and prompts timeout.  "source languageserver.golang timeout after 5000ms"

My guess is that the loading timeout is caused by too much code?

**Comment by oblitum** [comment, marked as answer]
If you search "timeout" in [schema.json](https://github.com/neoclide/coc.nvim/blob/master/data/schema.json) or in the contents of `:h coc-nvim.txt` you'll find timeout settings and their default values.

**Reply by leolin49** [reply]
thanks for your reply. 
I tried to change the timeout time to the maximum value of 15000ms in 'schema.json', but it still couldn't solve the problem completely. Not only that, this timeout error caused my CentOS server network to deteriorate.

**Reply by oblitum** [reply]
I'd recommend you to debug the server, it may be spinning or broken. Without that info it's hard to provide any further help.

- https://github.com/neoclide/coc.nvim/blob/master/.github/ISSUE_TEMPLATE/bug_report.md
- https://github.com/neoclide/coc.nvim/wiki/Debug-language-server
